db/system_keyspace.cc

```diff
@@ -2767,7 +2767,7 @@ future<service::topology> system_keyspace::load_topology_state() {
                     "load_topology_state: node {} in normal state but missing ring slice", host_id));
             }
         } else if (nstate == service::node_state::left) {
```

Comment:
> Those nodes will be kept in tablet replica sets for a while after node
replace is done, until the new replica is rebuilt. So we need to know
about those node's location (dc, rack) for two reasons:


> It's ok to not know the IP, and we don't keep it. Those nodes will not
be present in the IP-based replica sets, e.g. those returned by
get_natural_endpoints(), only in host_id-based replica
sets. storage_proxy request coordination is not affected.

> Nodes in the left state are still not present in token ring, and not
considered to be members of the ring (datacanter endpoints excludes them).


What happens to drivers? Do they get spooked by the inconsistency between the token ring and tablet routing information?

Reply:
@avelanarius @sylwiaszunejko ^

Reply:
Tablets get their replica information from this: https://github.com/scylladb/scylladb/blob/f83df24108851fa856bbc33e6d1e751d783ecdc2/locator/tablets.cc#L179 and it has replicas identified by host_id

Reply:
There should not be a problem. For example, in python-driver, it looks like that:
https://github.com/scylladb/python-driver/blob/master/cassandra/policies.py#L390
it selects hosts that are both in child policy plan and in the tablet_replicas. If there is a host id in tablet replicas that the driver doesn't know about, this host won't be matched with any host in child policy plan and therefore won't be used.